packages/components/src/public/QueryModel/EditableDetailPanel.tsx

```diff
@@ -122,117 +137,108 @@ export class EditableDetailPanel extends PureComponent<EditableDetailPanelProps,
+                            editingMode
+                            fileInputRenderer={fileInputRenderer}
+                            onAdditionalFormDataChange={onAdditionalFormDataChange}
+                            queryConfig={{
+                                ...model.queryConfig,
+                                // Issue 46478: Include update columns in request columns to ensure values are available
+                                requiredColumns: model.requiredColumns.concat(
+                                    model.updateColumns.map(col => col.fieldKey)
+                                ),
+                            }}
+                        />
+                    )}
                 </div>
             </div>
+        </div>
+    );
+
+    if (editing) {
+        return (
+            <Formsy
+                onChange={handleFormChange}
+                onValidSubmit={handleSubmit}
+                onValid={enableSubmitButton}
+                onInvalid={disableSubmitButton}
+            >
+                {panel}
+
+                <FormButtons>
+                    <button className="btn btn-default" type="button" onClick={toggleEditing}>
+                        Cancel
+                    </button>
+                    <CommentTextArea
+                        actionName="Update"
+                        containerClassName="inline-comment"
+                        onChange={_onCommentChange}
+                        requiresUserComment={requiresUserComment}
+                        inline
+                    />
+                    <button className="btn btn-success" type="submit" disabled={!canSubmit || (requiresUserComment && !hasValidUserComment) || disabled}>
+                        {submitText}
+                    </button>
+                </FormButtons>
+
+                {asSubPanel && <div className="panel-divider-spacing" />}
+            </Formsy>
         );
+    }
 
-        if (editing) {
-            return (
-                <Formsy
-                    onChange={this.handleFormChange}
-                    onValidSubmit={this.handleSubmit}
-                    onValid={this.enableSubmitButton}
-                    onInvalid={this.disableSubmitButton}
-                >
-                    {panel}
-
-                    <FormButtons>
-                        <button className="btn btn-default" type="button" onClick={this.toggleEditing}>
-                            Cancel
-                        </button>
-                        <button className="btn btn-success" type="submit" disabled={!canSubmit || disabled}>
-                            {submitText}
-                        </button>
-                    </FormButtons>
-
-                    {asSubPanel && <div className="panel-divider-spacing" />}
-                </Formsy>
-            );
-        }
+    return panel;
+}
 
-        return panel;
-    }
+EditableDetailPanel.defaultProps = {
+    api: getDefaultAPIWrapper(),
```

Comment:
Now that you've converted this to a `FC` you can use `const { api } = useAppContext();` to get the API wrappers instead.